### PR TITLE
python3Packages.tornado: 5.1 -> 6.0.4

### PR DESCRIPTION
**Pull request by marsam**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/tornado/default.nix [hunk 1]
```diff
@@ -28,9 +28,7 @@ buildPythonPackage rec {
   pname = "tornado";
   inherit version;
 
-  propagatedBuildInputs = [ backports_abc  certifi singledispatch ]
-    ++ lib.optional (pythonOlder "3.5") backports_ssl_match_hostname
-    ++ lib.optional (pythonOlder "3.2") futures;
+  propagatedBuildInputs = lib.optionals isPy27 [ backports_abc certifi singledispatch backports_ssl_match_hostname futures ];
```
Comment (jonringer) [inline pkgs/development/python-modules/tornado/default.nix:31]: please format
```suggestion
  propagatedBuildInputs = lib.optionals isPy27 [
    backports_abc
    certifi
    singledispatch
    backports_ssl_match_hostname
    futures
  ];
```

File: pkgs/development/python-modules/tornado/default.nix [hunk 2]
```diff
@@ -6,18 +6,18 @@
...
+, isPy27
+, version ? "6.0.4"
 }:
 
 let
   versionMap = {
     "4.5.3" = {
       sha256 = "02jzd23l4r6fswmwxaica9ldlyc2p6q8dk6dyff7j58fmdzf853d";
     };
-    "5.1" = {
```
Comment (jonringer) [inline pkgs/development/python-modules/tornado/default.nix:19]: we may want to keep around a v5 for applications which aren't ready to upgrade to 6

**Review by jonringer — COMMENTED** [review]
for a more graceful upgrade, maybe we would introduce `tornado_6` first, and slowly convert the packages over to that.

This would also cause 0 rebuilds

**Comment by FRidh** [comment]
This is going to need more testing.

**Comment by marsam** [comment]
addressed in https://github.com/NixOS/nixpkgs/commit/234239b57fbdd4eaca572cb2800b37f166ef2849

**Pull request closed without merging**